client/state/routing/middleware.js

```diff
@@ -3,50 +3,28 @@
  */
 import debugFactory from 'debug';
 import { isEmpty } from 'lodash';
-import page from 'page';
 
 /**
  * Internal dependencies
  */
-import {
-	getSavedPath,
-	savePath,
-} from 'lib/restore-last-path';
+import { savePath } from 'lib/restore-last-path';
 import { ROUTE_SET } from 'state/action-types';
 
 const debug = debugFactory( 'calypso:restore-last-location' );
-let hasInitialized = false;
 
 export const routingMiddleware = () => {
 	return ( next ) => ( action ) => {
-		if ( action.type !== ROUTE_SET || ! action.path || ! action.query ) {
+		if ( action.type !== ROUTE_SET ||
+				! action.path ||
+				! action.query ||
```

Comment:
I think we may be able to ommit `! action.query ||` here as that condition should be covered by `isEmpty`

From the [lodash REPL](https://lodash.com/docs/4.17.4#isEmpty)
```js
a = {
    b: [],
    c: [1]
}

_.isEmpty( a.d );
// => true
```